src/sentry/consumers/__init__.py

```diff
@@ -75,6 +76,10 @@ def multiprocessing_options(
 
 
 # consumer name -> consumer definition
+# XXX: default_topic is needed to lookup the schema even if the actual topic name has been
+# overridden. This is because the current topic override mechanism means the default topic name
+# is no longer available anywhere in code. We should probably fix this later so we don't need both
+#  "topic" and "default_topic" here though.
```

Comment:
I think we can fix this by defining a mapping in settings from logical topic to topic settings key, like so:

```
TOPIC_DEFINITIONS = {
    "snuba-generic-metrics": "KAFKA_SNUBA_GENERIC_METRICS"
}
```

Then we can have a new abstraction layer like `get_topic_definition2("snuba-generic-metrics")` that is used everywhere in code, and eventually we can make a breaking change in config format without too much effort.

Reply:
Yeah - that approach seems good to me. Added a task in our backlog for this. I considered adding it in this PR, but didn't want to get too off track here.